### ISSUE TITLE
Comments in decorators are moved to top when class is exported

### ISSUE THREAD

**Issue by mafredri** [prettier/prettier#1956]
Stumbled across this while formatting an Angular codebase.

**Input:**

```typescript
@NgModule({
  // Imports.
  imports: [
    // Angular modules.
    BrowserModule,

    // App modules.
    CoreModule,
    SharedModule,
  ],
})
export class AppModule {}
```

**Output:**

```typescript
// Imports.
// Angular modules.
// App modules.
@NgModule({
  imports: [BrowserModule, CoreModule, SharedModule],
})
export class AppModule {}
```

**Expected:**

No difference between input and output.

---

The problem goes away when `export` is removed from `export class AppModule {}`.

A reproduction of this issue can be found [here](https://prettier.github.io/prettier/#%7B%22content%22%3A%22%40NgModule(%7B%5Cn%20%20%2F%2F%20Imports.%5Cn%20%20imports%3A%20%5B%5Cn%20%20%20%20%2F%2F%20Angular%20modules.%5Cn%20%20%20%20BrowserModule%2C%5Cn%5Cn%20%20%20%20%2F%2F%20App%20modules.%5Cn%20%20%20%20CoreModule%2C%5Cn%20%20%20%20SharedModule%2C%5Cn%20%20%5D%2C%5Cn%7D)%5Cnexport%20class%20AppModule%20%7B%7D%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22es5%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D).

**Comment by azz** [comment]
Turns out this isn't TypeScript specific. [Happens with Babylon, too](https://prettier.github.io/prettier/#%7B%22content%22%3A%22%40NgModule(%7B%5Cn%20%20%2F%2F%20Imports.%5Cn%20%20imports%3A%20%5B%5Cn%20%20%20%20%2F%2F%20Angular%20modules.%5Cn%20%20%20%20BrowserModule%2C%5Cn%5Cn%20%20%20%20%2F%2F%20App%20modules.%5Cn%20%20%20%20CoreModule%2C%5Cn%20%20%20%20SharedModule%2C%5Cn%20%20%5D%2C%5Cn%7D)%5Cnexport%20class%20AppModule%20%7B%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D).

**Comment by mafredri** [comment]
Perhaps related, adding `// prettier-ignore` above the decorator removes it entirely, comments and class are preserved though. Again, removing `export` fixes the issue.

**Comment by vjeux** [comment]
I just published 1.4.3 which fixes it. Thanks for reporting it!
